master/internal/api_auth.go

```
@@ -152,14 +170,32 @@ func processProxyAuthentication(c echo.Context) (done bool, err error) {
 	// Continue NTSC task checks.
 	if spec.TaskType == model.TaskTypeTensorboard {
 		err = command.AuthZProvider.Get().CanGetTensorboard(
-			ctx, *user, spec.WorkspaceID, spec.ExperimentIDs, spec.TrialIDs)
+			ctx, *usr, spec.WorkspaceID, spec.ExperimentIDs, spec.TrialIDs)
 	} else {
 		err = command.AuthZProvider.Get().CanGetNSC(
-			ctx, *user, spec.WorkspaceID)
+			ctx, *usr, spec.WorkspaceID)
 	}
 	return err != nil, authz.SubIfUnauthorized(err, serviceNotFoundErr)
 }
 
+// extractNotebookTokenFromRequest looks for auth token for Jupyter notebooks
+// in two places:
+// 1. A token query parameter in the request URL.
+// 2. An HTTP Authorization header with a "token" type.
+func extractNotebookTokenFromRequest(r *http.Request) (string, error) {
```

maybe this doesn't need to return an error?